plugins/generator-1.19.4/forge-1.19.4/templates/livingentity/livingentity.java.ftl

```diff
@@ -222,6 +238,79 @@ public class ${name}Entity extends ${extendsClass} <#if data.ranged>implements R
    	}
 	</#if>
 
+	<#if data.canTrade>
+	protected void updateTrades() {
+		<#assign professions = (w.filterBrokenReferences(data.tradeProfessions))?filter(p -> p.getUnmappedValue() != "WANDERING_TRADER")>
+		List<VillagerProfession> professions = List.of(
+			<#list professions as profession>
+				${profession}<#if profession?has_next>,</#if>
+			</#list>
+		);
+		Int2ObjectMap<VillagerTrades.ItemListing[]> trades = new Int2ObjectOpenHashMap<>();
+		VillagerTrades.TRADES.forEach((key, value) -> {
+			if (<#if data.tradeProfessionsExclude>!</#if>(professions.contains(key))) {
+				value.int2ObjectEntrySet().forEach(ent -> trades.put(ent.getIntKey(), Arrays.copyOf(ent.getValue(), ent.getValue().length)));
+			}
+		});
+		<#if data.tradeProfessions?contains("WanderingTrader") && !data.tradeProfessionsExclude>
+		VillagerTrades.WANDERING_TRADER_TRADES.int2ObjectEntrySet().forEach(e -> trades.put(e.getIntKey(), Arrays.copyOf(e.getValue(), e.getValue().length)));
+		</#if>
+		if (trades != null && !trades.isEmpty()) {
+			VillagerTrades.ItemListing[] leveledTrades = trades.get(<#if data.villagerTradingType>this.getVillagerData().getLevel()<#else>1</#if>);
+			if (leveledTrades != null) {
+				this.addOffersFromItemListings(this.getOffers(), leveledTrades, 2);
+			}
+		}
+	}
+
+	<#if data.tradeRestockCondition?? && (hasProcedure(data.tradeRestockCondition) || data.tradeRestockCondition.getFixedValue())>
+	<#if data.villagerTradingType>@Override</#if> public boolean canRestock() {
+		<#if hasProcedure(data.tradeRestockCondition)>
+			Entity entity = this;
+			Level world = entity.level;
+			double x = entity.getX();
+			double y = entity.getY();
+			double z = entity.getZ();
+		</#if>
+		return <@procedureOBJToConditionCode data.tradeRestockCondition true false/>;
+	}
+	</#if>
+
+	<#if data.villagerTradingType>
+	@Override ${mcc.getMethod("net.minecraft.world.entity.npc.Villager", "customServerAiStep")
+		.replace("this.stopTrading();", "")}
+
+	private void updateSpecialPrices(Player player) {

```

Comment:
@Defeatomizer some help on this plz?

Reply:
You'd need to modify access transformer to open access to this method in `Villager` class so that you can override it here, something like:
```freemarker
${method?then("public", "public-f")} ${pathToClass} <#if method>m_${hash}_()${argumentTypes?join("")}<#else>f_${hash}_</#if> #${mappedName}${method?then("()", "")}
```
where `hash` needs to be found in the unmapped source code. In your case, this "template" turns into:
```freemarker
public net.minecraft.world.entity.npc.Villager m_35540_()Lnet/minecraft/world/entity/player/Player; #updateSpecialPrices()
```

Reply:
It does not work. I tested and `updateSpecialPrices` is still private even after reloading ATs

Reply:
Are you sure the method name `m_35540_` is correct? Same for other arguments?

Reply:
Maybe it appears private in code, I would recommend trying whether overriding the method is possible to make sure whether access rights were extended

Reply:
This does not make sense. Aren't methods made public through ATs appear public in the code as well?

Reply:
No. This is a configuration file for unlocking methods/fields **at runtime**.

Reply:
Then what's the point of making it public if I can't override it? What am I doing wrong?

Reply:
Klemen, would removing `@Override` annotation from this single method fix it?

Reply:
No, if override does not work, it means the method is not being overridden, thus method from parent is used, which is not ok as wrong profession data is used

Reply:
An easier solution would be to copy all methods that use that private method in this case?

Reply:
AT is a much cleaner solution. This PR is already quite messy and copying all that code would do it no favor

Reply:
Maybe I did mess the transformer up and not all of these are correct?
```
public net.minecraft.world.entity.npc.Villager m_35540_()Lnet/minecraft/world/entity/player/Player; #updateSpecialPrices()
```

Reply:
I don't have time RN to check unfortunately

Reply:
> Maybe I did mess the transformer up and not all of these are correct?
> 
> ```
> public net.minecraft.world.entity.npc.Villager m_35540_()Lnet/minecraft/world/entity/player/Player; #updateSpecialPrices()
> ```

You did, here's the correct one :
```
public net.minecraft.world.entity.npc.Villager m_35540_(Lnet/minecraft/world/entity/player/Player;)V #updateSpecialPrices()
```

Reply:
It's done!

Reply:
So the template should be:
```freemarker
${method?then("public", "public-f")} ${pathToClass} <#if method>m_${hash}_(${argumentTypes?join("")})${returnType}<#else>f_${hash}_</#if> #${mappedName}${method?then("()", "")}
```